packages/shared/lib/clients/big-query.client.ts

```diff
@@ -0,0 +1,68 @@
+import { BigQuery } from '@google-cloud/bigquery';
+import logger from '../logger/console.js';
+import { isCloud, getEnv } from '../utils/utils.js';
+
+type Row = Record<string, string | number | undefined>;
+
+class BigQueryClient {
+    private client: BigQuery;
+    private datasetName: string;
+    private tableName: string;
+
+    constructor() {
+        this.client = new BigQuery();
+        this.datasetName = 'usage';
+        this.tableName = 'scriptRuns';
+    }
+
+    private async createDataSet() {
+        const dataset = this.client.dataset(this.datasetName);
+        const [exists] = await dataset.exists();
+        if (!exists) {
+            await this.client.createDataset(this.datasetName);
+        }
+    }
+
+    private async createTable() {
+        const table = this.client.dataset(this.datasetName).table(`${getEnv()}_${this.tableName}`);
+        const [exists] = await table.exists();
+        if (!exists) {
+            await table.create({
+                schema: {
+                    fields: [
+                        { name: 'executionType', type: 'STRING' },
+                        { name: 'connectionId', type: 'STRING' },
```

Comment:
@khaliqgant which connection ID is this? The one picked by the user, or the primary key from the db? We probably want both.

Reply:
Added `internalConnectionId` which is the primary key from the db

Reply:
Thank you!